py/src/braintrust/logger.py

```diff
@@ -5193,9 +5195,21 @@ def submit(self, fn: Callable[..., Any], *args: Any, **kwargs: Any) -> Any:
         # Capture all current context variables
         context = contextvars.copy_context()
 
+        # Capture the current thread-local background logger override
+        # This is necessary because worker threads have their own thread-local storage
+        override_logger = getattr(_state._override_bg_logger, "logger", None)
+
         def wrapped_fn(*args, **kwargs):
-            # Run the function inside the captured context
-            return context.run(fn, *args, **kwargs)
+            # Restore the background logger override in the worker thread
+            if override_logger is not None:
```

Comment:
i wonder if there's an way to make this non-specific t othe override logger ... are there other state variables that need ot be copied?

Reply:
https://github.com/braintrustdata/braintrust-sdk/blob/main/py/src/braintrust/logger.py#L460-L477

I guess I was thinking that maybe there was other state that might need to be copied as well. I wasn't sure if copying the background logger only was enough?